### PR TITLE
[C-2121] Offline downloads settings

### PR DESCRIPTION
**Pull request by amendelsohn**
### Description

Based on changes from [`dj-cellular-download-option` ](https://github.com/AudiusProject/audius-client/tree/dj-cellular-download-option) thanks @dylanjeffers 👏 

Adds settings page for downloads
Also minor settings redesign

<p float="left">

</p>

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/amendel-cellular-download-option

File: packages/mobile/src/store/offline-downloads/sagas/offlineQueueSagas/watchNetworkType.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  requestProcessNextJob,
+  setCurrentNetworkType,
+  setDownloadNetworkPreference,
+  updateQueueStatus
+} from '../../slice'
+
+const { SET_UNREACHABLE, SET_REACHABLE } = reachabilityActions
+const { getIsReachable } = reachabilitySelectors
+
+export function* watchNetworkType() {
```
Comment (dylanjeffers) [inline packages/mobile/src/store/offline-downloads/sagas/offlineQueueSagas/watchNetworkType.ts:22]: hell yeah

File: packages/mobile/src/store/offline-downloads/sagas/utils/shouldCancelJob.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  getDownloadNetworkTypePreference
+} from '../../selectors'
+import {
+  setCurrentNetworkType,
+  setDownloadNetworkPreference
+} from '../../slice'
+const { SET_UNREACHABLE } = reachabilityActions
+const { getIsReachable } = reachabilitySelectors
+
+export function* shouldCancelJob() {
```
Comment (dylanjeffers) [inline packages/mobile/src/store/offline-downloads/sagas/utils/shouldCancelJob.ts:16]: love this omg!

**Review by dylanjeffers — APPROVED** [review]
so fresh!!! all the code is so clean and nice great job! love how you extended buttons, and refactored the settings rows etc. also love the shouldCancelJob stuff too

File: packages/mobile/src/screens/settings-screen/RemoveAllDownloadsRow.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  }
+}))
+
+export const RemoveAllDownloadsRow = () => {
+  const isOfflineDownloadEnabled = useIsOfflineModeEnabled()
+  const dispatch = useDispatch()
+  const styles = useStyles()
+
+  const offlineCollectionStatus = useSelector(getOfflineCollectionsStatus)
+  const isAnyCollectionDownloaded =
```
[the remark this answers is not in the document]
Comment (amendelsohn) [reply packages/mobile/src/screens/settings-screen/RemoveAllDownloadsRow.tsx:33]: yeah wasn't sure it was worth it for this one, but I think given the pattern we're trying to set, it makes sense to do it here too

File: packages/mobile/src/store/offline-downloads/sagas/offlineQueueSagas/watchNetworkType.ts (hunk 1 above)
Comment (amendelsohn) [reply packages/mobile/src/store/offline-downloads/sagas/offlineQueueSagas/watchNetworkType.ts:22]: you wrote this 🤣 but yeah kudos!

File: packages/mobile/src/store/offline-downloads/sagas/utils/shouldCancelJob.ts (hunk 2 above)
Comment (amendelsohn) [reply packages/mobile/src/store/offline-downloads/sagas/utils/shouldCancelJob.ts:16]: was this not also you? lol
I based all this off your branch `dj-cellular-download-option`

Comment (dylanjeffers) [reply packages/mobile/src/store/offline-downloads/sagas/utils/shouldCancelJob.ts:16]: wait really? omg it was so long ago

File: packages/mobile/src/store/offline-downloads/sagas/offlineQueueSagas/watchNetworkType.ts (hunk 1 above)
Comment (dylanjeffers) [reply packages/mobile/src/store/offline-downloads/sagas/offlineQueueSagas/watchNetworkType.ts:22]: yoo lol self ownage

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

**Pull request merged**